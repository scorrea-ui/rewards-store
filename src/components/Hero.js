import React from 'react'

export const Hero = ({ title, background }) => {
  return (
    <div
      className='c-hero'
      style={{
        backgroundImage: background
          ? `url(${background})`
          : `url(${process.env.PUBLIC_URL}/header.png)`,
      }}
    >
      <div className='o-container'>
        <div className='o-grid'>
          <h1 className='c-hero__title'>{title}</h1>
        </div>
      </div>
    </div>
  )
}
