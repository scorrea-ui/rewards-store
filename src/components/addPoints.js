import React from "react";
import { addUserPoints } from "./redux/actions";
import { connect } from "react-redux";
import ButtonPoints from "./ButtonPoints";

const AddPoints = ({ addPoints }) => {
  return (
    <>
      <div className="o-container">
        <div className="o-grid o-grid--center o-grid--middle">
          <div className="o-grid__col u-4/12@md">
            <ButtonPoints
              className="c-btn--primary"
              handler={addPoints}
              amount="1000"
            />
          </div>
          <div className="o-grid__col u-4/12@md">
            <ButtonPoints
              className="c-btn--secondary"
              handler={addPoints}
              amount="5000"
            />
          </div>
          <div className="o-grid__col u-4/12@md">
            <ButtonPoints
              className="c-btn--tertiary"
              handler={addPoints}
              amount="7500"
            />
          </div>
        </div>
      </div>
    </>
  );
};

const mapStateToProps = (state) => {
  return {
    points: state.points,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    addPoints: (amount) => dispatch(addUserPoints(amount)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(AddPoints);
