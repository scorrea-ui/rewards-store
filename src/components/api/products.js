import { headers } from "./utils/headers";

const products = {
  get: {
    url: "https://coding-challenge-api.aerolab.co/user/history",
    method: "GET",
  },
  redeem: {
    url: "https://coding-challenge-api.aerolab.co/redeem",
    method: "POST",
  },
};

export const fetchProducts = () => {
  return fetch(products.get.url, {
    method: products.get.method,
    headers,
  });
};

export const redeemPoints = (id) => {
  return fetch(products.redeem.url, {
    method: products.redeem.method,
    headers,
    body: JSON.stringify({ productId: id }),
  });
};
