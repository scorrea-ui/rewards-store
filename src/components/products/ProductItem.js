import React from "react";

const ProductItem = ({ category, cost, img, name, redeemPoints, points }) => {
  return (
    <div className="o-grid__col u-3/12@md">
      <div className="c-product__box">
        <img
          className="c-product__img"
          src={
            !img
              ? img
              : process.env.PUBLIC_URL + "/product-pics/MacbookPro-x1.png"
          }
          alt={name}
        />
        <hr />
        <div className="c-product__content">
          <p className="c-product__category">{category}</p>
          <p className="c-product__name">{name}</p>
        </div>
        {points > cost ? (
          <div className="c-product__reveal">
            <>
              <div className="c-product__coins">
                <span>{cost}</span>
                <img
                  src={process.env.PUBLIC_URL + "/icons/coin.svg"}
                  alt="Coins"
                ></img>
              </div>
              <button className="c-product__button" onClick={redeemPoints}>
                Redeem now
              </button>
            </>
          </div>
        ) : null}
        {points > cost ? (
          <div className="c-product__buy c-product__buy--success"></div>
        ) : (
          <div className="c-product__buy c-product__buy--error">
            <span>Missing {cost - points}</span>
            <img
              src={process.env.PUBLIC_URL + "/icons/coin.svg"}
              alt="Coins"
            ></img>
          </div>
        )}
      </div>
    </div>
  );
};

export default ProductItem;
