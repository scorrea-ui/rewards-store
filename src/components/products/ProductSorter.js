import React from "react";

export const ProductSorter = ({ sortProducts, products, title, sorter }) => {
  return (
    <button className="c-btn c-btn--sorters" onClick={() => sortProducts(products, sorter)}>{title}</button>
  );
};
