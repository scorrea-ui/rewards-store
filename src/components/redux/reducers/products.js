import {
  GET_PRODUCTS_PENDING,
  GET_PRODUCTS_FAILED,
  GET_PRODUCTS_SUCCESS,
  ORDER_PRODUCTS_BY_PRICE,
  ORDER_PRODUCTS_BY_CATEGORY,
} from "../actionTypes";

const initialState = {
  products: [],
  loading: false,
  error: null,
  points: null,
  message: "",
  sort: "",
};

export default function (state = initialState, action) {
  switch (action.type) {
    case GET_PRODUCTS_PENDING: {
      return {
        ...state,
        loading: true,
      };
    }
    case GET_PRODUCTS_SUCCESS: {
      return {
        ...state,
        products: action.payload,
        loading: false,
      };
    }
    case GET_PRODUCTS_FAILED: {
      return {
        ...state,
        error: action.error,
        loading: false,
      };
    }

    case ORDER_PRODUCTS_BY_PRICE: {
      return {
        ...state,
        products: action.payload,
        sort: action.payload.sort,
      };
    }

    case ORDER_PRODUCTS_BY_CATEGORY: {
      return {
        ...state,
        products: action.payload,
        sort: action.payload.sort,
      };
    }

    default: {
      return state;
    }
  }
}
