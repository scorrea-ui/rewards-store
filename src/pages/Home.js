import React from "react";
import Header from "../components/global/Header";
import { Hero } from "../components/Hero";
import Products from "../components/products/Products";

export const Home = () => {
  return (
    <div>
      <Header />
      <Hero title="Electronics" />
      <Products />
    </div>
  );
};
