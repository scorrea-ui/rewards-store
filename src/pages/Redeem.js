import React from "react";
import Header from "../components/global/Header";
import { Hero } from "../components/Hero";
import History from "../components/history/History";

export const Redeem = () => {
  return (
    <div>
      <Header />
      <Hero title="Historial de productos" background={process.env.PUBLIC_URL + '/rewards.jpg'} />
      <History />
    </div>
  );
};
