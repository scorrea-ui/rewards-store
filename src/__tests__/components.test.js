import React from 'react'
import { mount, shallow, render } from 'enzyme'
import { Home } from '../pages/Home'
import { Hero } from '../components/Hero'
import { Redeem } from '../pages/Redeem'
import ProductsHeader from '../components/products/productsHeader'
import { ProductSorter } from '../components/products/ProductSorter'
import ProductItem from '../components/products/ProductItem'

describe('Home Page', () => {
  it('should render Home Page correctly', () => {
    const component = shallow(<Home />)
    expect(component).toMatchSnapshot()
  })
})

describe('Redeem Page', () => {
  it('should render the Redeem page correctly', () => {
    const component = shallow(<Redeem />)
    expect(component).toMatchSnapshot()
  })
})

describe('Has className', () => {
  it('should render the product sorter component with class of c-btn--sorters', () => {
    const component = mount(<ProductSorter />)
    expect(component.find('.c-btn').hasClass('c-btn--sorters')).toEqual(true)
  })
})

describe('Has Props', () => {
  it('should render Hero correctly with given title', () => {
    const title = 'Hero'
    const component = shallow(<Hero title={title} />)
    expect(component).toMatchSnapshot()
  })

  it('checks the type of value', () => {
    const props = {
      cost: '1000',
    }
    const component = mount(<ProductItem {...props} />)
    expect(typeof component.prop('cost')).toBe('string')
  })
})
