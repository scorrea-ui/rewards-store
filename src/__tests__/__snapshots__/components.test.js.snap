// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Has Props should render Hero correctly with given title 1`] = `
<div
  className="c-hero"
>
  <div
    className="o-container"
  >
    <div
      className="o-grid"
    >
      <h1
        className="c-hero__title"
      >
        Hero
      </h1>
    </div>
  </div>
</div>
`;

exports[`Hero Component should render Hero correctly with given title 1`] = `
<div
  className="c-hero"
>
  <div
    className="o-container"
  >
    <div
      className="o-grid"
    >
      <h1
        className="c-hero__title"
      >
        Hero
      </h1>
    </div>
  </div>
</div>
`;

exports[`Hero should render with title props should render Hero correctly with given title 1`] = `
<div
  className="c-hero"
>
  <div
    className="o-container"
  >
    <div
      className="o-grid"
    >
      <h1
        className="c-hero__title"
      >
        Hero
      </h1>
    </div>
  </div>
</div>
`;

exports[`Home Page should render Home Page correctly 1`] = `
<div>
  <Connect(Header) />
  <Hero
    title="Lista de productos"
  />
  <Connect(Products) />
</div>
`;

exports[`MyComponent should render Home Page correctly 1`] = `
<div>
  <Connect(Header) />
  <Hero
    title="Lista de productos"
  />
  <Connect(Products) />
</div>
`;

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div>
  <Connect(Header) />
  <Hero
    title="Lista de productos"
  />
  <Connect(Products) />
</div>
`;

exports[`Redeem Page should render the Redeem page correctly 1`] = `
<div>
  <Connect(Header) />
  <Hero
    title="Historial de productos"
  />
  <Connect(History) />
</div>
`;

exports[`should render Hero correctly with given title 1`] = `
<div
  className="c-hero"
>
  <div
    className="o-container"
  >
    <div
      className="o-grid"
    >
      <h1
        className="c-hero__title"
      >
        Hero
      </h1>
    </div>
  </div>
</div>
`;
